File: tests/__tests__/binary_search_trees/utility.js
import { BST } from "@/binary_search_trees/assets/BST.js";

export const buildBST = (nodes, rootId) => {
  const idToNode = {};

  nodes.forEach((node) => {
    idToNode[node.id] = new BST(node.value);
  });

  nodes.forEach((node) => {
    const bstNode = idToNode[node.id];
    bstNode.left = node.left ? idToNode[node.left] : null;
    bstNode.right = node.right ? idToNode[node.right] : null;
  });

  return idToNode[rootId];
};

File: tests/python/binary_search_trees/utility.py
from problems.binary_search_trees.assets.BST import BST


def build_bst(nodes: list[dict], root_id: str) -> BST:
    id_to_node = {}

    for node in nodes:
        id_to_node[node["id"]] = BST(node["value"])

    for node in nodes:
        bst_node = id_to_node[node["id"]]
        bst_node.left = id_to_node.get(node["left"])
        bst_node.right = id_to_node.get(node["right"])

    return id_to_node[root_id]
